tembo-operator/src/cloudnativepg/hibernate.rs

```diff
@@ -337,6 +277,134 @@ pub async fn reconcile_cluster_hibernation(cdb: &CoreDB, ctx: &Arc<Context>) ->
     Ok(())
 }
 
+/// Patches AppService deployments in a Kubernetes cluster by updating their replica count based on the CoreDB specification.
+///
+/// This function performs the following operations:
+/// * Updates the replica count to 0 if CoreDB is stopped, or 1 if it's running
+/// * Deletes associated PodMonitor resources when scaling down to 0 replicas
+/// * Applies patches to all relevant deployments in the specified namespace
+///
+/// # Arguments
+///
+/// * `ctx` - A shared reference to the Context containing the Kubernetes client
+/// * `namespace` - The Kubernetes namespace where the deployments exist
+/// * `name` - The name identifier for the cluster
+/// * `cdb` - Reference to the CoreDB specification containing the stop status
+///
+/// # Returns
+///
+/// * `Ok(())` if all patches were successfully applied
+/// * `Err(Action)` if any operation fails, triggering a requeue with jitter
+///
+/// # Errors
+///
+/// This function will return an error in the following situations:
+/// * Failed to retrieve deployment list
+/// * Failed to delete PodMonitor when scaling down
+/// * Failed to apply deployment patches
+async fn patch_appservice_deployments(
+    ctx: &Arc<Context>,
+    namespace: &str,
+    name: &str,
+    cdb: &CoreDB,
+) -> Result<(), Action> {
+    let client = ctx.client.clone();
+    let deployment_api: Api<Deployment> = Api::namespaced(client.clone(), namespace);
+    let podmonitor_api: Api<podmon::PodMonitor> = Api::namespaced(client.clone(), namespace);
+    let ps = PatchParams::apply("patch_merge").force();
+
+    let replicas = if cdb.spec.stop { 0 } else { 1 };
+    let replica_patch = json!({
+        "apiVersion": "apps/v1",
+        "kind": "Deployment",
+        "spec": {
+            "replicas": replicas,
+        }
+    });
+
+    let deployment_list = match get_appservice_deployment_objects(&client, namespace, name).await {
+        Ok(deployments) => deployments,
+        Err(e) => {
+            warn!(
+                "Could not retrieve deployment list for cluster {}; retrying",
+                name
+            );
+            debug!("Caught error {}", e);
+            return Err(requeue_normal_with_jitter());
+        }
+    };
+
+    for deployment in deployment_list {
+        let spec = match deployment.spec {
```

Comment:
Tip for next time:

```rust
let Some(spec) = deployment.spec else {
    continue;
}
```

Reply:
I went ahead and just changed it to your suggestion.  Thanks!